python/ray/data/preprocessors/vectorizer.py

```diff
@@ -9,15 +9,84 @@
+        dislike Python"` might have the document-term matrix below:
+
+        .. code-block::
+
+                    corpus_I  corpus_Python  corpus_dislike  corpus_like
+                0         1              1               1            0
+                1         1              1               0            1
+
+        To generate the matrix, you need to map each token to a unique index. For example:
+
+        .. code-block::
+
+                        token  index
+                0        I      0
+                1   Python      1
+                2  dislike      2
+                3     like      3
+
+        The problem with this approach is that the mapping and consequently memory use
+        scale linearly with the size of your vocabulary.
+
+        :class:`HashingVectorizer` circumvents this problem by computing
+        indices with a hash function: :math:`\\texttt{index} = hash(\\texttt{token})`.
 
     Args:
-        columns: The columns that will individually be hashed.
-        num_features: The size of each hashed feature vector.
-        tokenization_fn: The tokenization function to use.
-            If not specified, a simple ``string.split(" ")`` will be used.
+        columns: The columns to separately tokenize and count.
+        num_features: The number of features used to represent the vocabulary. You
+            should choose a value large enough to prevent hash collisions between
+            distinct tokens. :math:`2^{18}` is typically large enough for text-classification
+            problems.
+        tokenization_fn: The function used to generate tokens. This function
+            should accept a string as input and return a list of tokens as
+            output. If unspecified, the tokenizer uses a function equivalent to
+            ``lambda s: s.split(" ")``.
+
+    Examples:
+        >>> import pandas as pd
+        >>> import ray
+        >>> from ray.data.preprocessors import HashingVectorizer
+        >>>
+        >>> df = pd.DataFrame({
+        ...     "corpus": [
+        ...         "Jimmy likes volleyball",
+        ...         "Bob likes volleyball too",
+        ...         "Bob also likes fruit jerky"
+        ...     ]
+        ... })
+        >>> ds = ray.data.from_pandas()
+        >>>
+        >>> vectorizer = HashingVectorizer(["corpus"], num_features=8)
+        >>> vectorizer.fit_transform(ds).to_pandas()
+                hash_corpus_0  hash_corpus_1  hash_corpus_2  hash_corpus_3  hash_corpus_4  hash_corpus_5  hash_corpus_6  hash_corpus_7
+        0              1              0              1              0              0              0              0              1
+        1              1              0              1              0              0              0              1              1
+        2              0              0              1              1              0              2              1              0
```

Comment:
Worth having an example that has values > 1? Either repeat or hash collision.

Reply:
There is a value > 1. Row 2, hash_corpus_5.